### PR TITLE
8.0 enhanced balance determination

### PR DESCRIPTION
**Pull request by NL66278**
- Support computing start balance from end balance
- More reliable detection of start and end balance
- Support CLAV type for end balance (used by German Postbank)
- Streamlined code for node parsing

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Another one that is conflicting and maybe not relevant anymore? Can you confirm?

**Comment by NL66278** [comment]
@pedrobaeza I will resolve the conflict. Still very relevant for one of our customers. And thereight be more people still on 8.0

**Comment by pedrobaeza** [comment]
OK then.

File: .travis.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ virtualenv:
 install:
   - git clone https://github.com/OCA/maintainer-quality-tools.git ${HOME}/maintainer-quality-tools
   - export PATH=${HOME}/maintainer-quality-tools/travis:${PATH}
-  - pip install ofxparse
+  - pip install git+https://github.com/jseutter/ofxparse.git
```
Comment (pedrobaeza) [inline .travis.yml:31]: Can you do this in requirements.txt if still needed?

**Review by pedrobaeza — COMMENTED** [review]
About unneeded changes for reducing the diff

File: .travis.yml (hunk 1 above)
Comment (NL66278) [reply .travis.yml:31]: @pedrobaeza But this already was in .travis.yml. I just updated the link to a working version. But if you think it better to create a new requirements.txt in the repository, I will do it.

Comment (pedrobaeza) [reply .travis.yml:31]: OK, no special problem. It's just the new convention for being clearer.

**Review by pedrobaeza — APPROVED** [review]
Code review

@StefanRijnhart @hbrunn @astirpe any of you using CAMT files can confirm these changes are correct?

**Review by thomaspaulb — APPROVED** [review]

**Comment by NL66278** [comment]
@dreispt Maybe you can try again? Merge failed because db server could not be reached. Sounds as a temporary problem.

**Comment by NL66278** [comment]
Ocabot is broken for these old branches. I would merge by hand if I could do this in this repository. @dreispt Can you merge this?

**Comment by StefanRijnhart** [comment]
```
OperationalError: could not connect to server: No such file or directory

	Is the server running locally and accepting

	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
```

It's the postgresql version that is now required in the travis configuration file.

See https://github.com/OCA/bank-statement-import/commit/65d882b1ae91408d087e3c23fd4f9cb1eb3dcbf4 from the 10.0 branch.

**Comment by NL66278** [comment]
@StefanRijnhart Thanks for the pointer. I updated .travis.yml in this PR as wel..

**Comment by NL66278** [comment]
@StefanRijnhart And do you have the rights in this repo to merge?

**Comment by StefanRijnhart** [comment]
Let's see if it works

/ocabot merge minor

**Pull request merged**